app.rb

```diff
@@ -0,0 +1,179 @@
+require_relative './student'
+require_relative './book'
+require_relative './teacher'
+require_relative './rental'
+require 'date'
+class App
+  attr_accessor :books, :people, :rentals
+
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def run
+    puts 'Welcome to Starminds School Library App!'
+    puts ''
+    puts ''
+    puts 'Please put an option by entering a number!'
+    until pick_option
+      input = gets.chomp
+      if input == '7'
+        puts 'Thank you for using this app'
+        break
+      end
+      run_option(input)
+    end
+  end
+
+  def pick_option
+    puts '1 - List all books'
+    puts '2 - List all People'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - list all rentals for a given person id'
+    puts '7 - Exit'
+  end
+
+  def run_option(input)
+    case input
+    when '1'
+      list_of_books
+    when '2'
+      list_of_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_of_rentals
+    else
+      'Invalid Input. Enter digit from 1 to 7'
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [input the number]'
+    select_person = gets.chomp.to_i
+    case select_person
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts 'invalid input. Input either 1 or 2 according to your choice'
+    end
+  end
```

Comment:
- [ ]  Good job so far but you still have some issues that need to be fixed. When I try to create a person (student or teacher) the console app failed with the message `wrong number of arguments`. Please kindly fix that 🙃 🙃

<img width="1168" alt="Screen Shot 2023-01-07 at 18 41 05" src="https://user-images.githubusercontent.com/74501523/211158945-36f8ed76-b7d4-4c1d-a477-a40e2620c571.png">

- [ ] Please after solving the bug you will make sure that the app works correctly 🙃 😉